### PR TITLE
[Dependencies] - Sort dependencies by type and path

### PR DESCRIPTION
**Pull request by BlackbitNeueMedien**
One of our clients is working with the dependencies on the objects and asked if it is possible to sort the list of data alphabetically by path - without clicking the path column on every page to sort (which of course only sorts for current page).

I customized the SQL query a bit so that JOIN can be used to find the path of the other tables (document, assets and objects).

### CODE REVIEW

**Comment by BlackbitNeueMedien** [comment]
Perhaps the same would make sense for https://github.com/pimcore/pimcore/blob/3d6d51756e678e1dd257573bec0404472cc9e2ec/models/Dependency/Dao.php#L47

File: tests/Model/Element/DependenciesTest.php [hunk 1]
```diff
@@ -78,7 +78,9 @@ public function testObjectDependencies()
         $source = TestHelper::createEmptyObject();
 
         /** @var Unittest[] $targets */
-        $targets = TestHelper::createEmptyObjects('', true, 3);
+        for ($i = 0; $i <= 2; $i++) {
+            $targets[] = TestHelper::createEmptyObjects($i, true, 1)[0];
```
Comment (dvesh3) [inline tests/Model/Element/DependenciesTest.php:82]:
```suggestion
            $targets[] = TestHelper::createEmptyObject($i, true);
```

File: tests/Model/Element/DependenciesTest.php [hunk 2]
```diff
@@ -103,7 +105,9 @@ public function testDocumentDependencies()
         /** @var Document $source */
         $source = TestHelper::createEmptyDocumentPage();
         /** @var Unittest[] $targets */
-        $targets = TestHelper::createEmptyObjects('', true, 3);
+        for ($i = 0; $i <= 2; $i++) {
+            $targets[] = TestHelper::createEmptyObjects($i, true, 1)[0];
```
Comment (dvesh3) [inline tests/Model/Element/DependenciesTest.php:109]:
```suggestion
            $targets[] = TestHelper::createEmptyObject($i, true);
```

File: tests/Model/Element/DependenciesTest.php [hunk 3]
```diff
@@ -128,7 +132,11 @@ public function testAssetDependencies()
         /** @var Asset $source */
         $source = TestHelper::createImageAsset();
         /** @var Unittest[] $targets */
-        $targets = TestHelper::createEmptyObjects('', true, 3);
+        $targets = [];
+        for($i=0;$i<=2;$i++) {
+            $targets[] = TestHelper::createEmptyObjects($i, true, 1)[0];
```
Comment (dvesh3) [inline tests/Model/Element/DependenciesTest.php:137]:
```suggestion
            $targets[] = TestHelper::createEmptyObject($i, true);
```

File: tests/Model/Element/DependenciesTest.php (hunk 1 above)
Comment (BlackbitNeueMedien) [reply tests/Model/Element/DependenciesTest.php]: Fixed.

File: tests/Model/Element/DependenciesTest.php (hunk 2 above)
Comment (BlackbitNeueMedien) [reply tests/Model/Element/DependenciesTest.php]: Fixed.

File: tests/Model/Element/DependenciesTest.php (hunk 3 above)
Comment (BlackbitNeueMedien) [reply tests/Model/Element/DependenciesTest.php]: Fixed.

**Comment by dvesh3** [comment]
thank you!

**Pull request merged**